src/main/java/mate/academy/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.security;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException(email));
```

Comment:
provide informative message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException(email));
```